app.rb

```diff
@@ -24,31 +24,39 @@ def list_people
     puts ''
   end
 
-  def create_person # rubocop:disable Metrics/MethodLength
+  def create_person
     puts 'Which type of person you wish to create'
     puts '1. Student'
     puts '2. Teacher'
     print 'Enter selection: '
     person_type = gets.chomp.to_i
     print 'Name: '
-    name = gets.chomp
+    @name = gets.chomp
     print 'Age : '
-    age = gets.chomp.to_i
+    @age = gets.chomp.to_i
     case person_type
     when 1
-      print 'Has parent permission? [Y/N]: '
-      permission = gets[0]
-      permission = (permission == ('Y' || 'y'))
-      @persons << Student.new('Unkown', age, name, permission)
-      puts "Person created successfully \n\n"
+      create_student
     when 2
-      print 'Specialization: '
-      specialization = gets.chomp
-      @persons << Teacher.new(specialization, age, name)
-      puts "Person created successfully\n\n"
+      create_teacher
     end
   end
 
+  def create_student
+    print 'Has parent permission? [Y/N]: '
+    permission = gets[0]
+    permission = (permission == ('Y' || 'y'))
```

Comment:
- [ ] Once more, let's kindly make sure that this method's only responsibility is to create a student. There should be NO requests for inputs from the user and input validation. Scroll up :point_up: and learn more from the previous comment.